UnitsNet.Serialization.JsonNet/UnitsNet.Serialization.JsonNet.csproj

```diff
@@ -27,18 +27,6 @@
     <TargetFramework>netstandard2.0</TargetFramework>
   </PropertyGroup>
 
-  <!-- SourceLink: https://github.com/dotnet/sourcelink -->
-  <PropertyGroup>
-    <PublishRepositoryUrl>true</PublishRepositoryUrl>
-
-    <!-- Optional: Embed source files that are not tracked by the source control manager in the PDB -->
-    <EmbedUntrackedSources>true</EmbedUntrackedSources>
-
-    <!-- Optional: Build symbol package (.snupkg) to distribute the PDB containing Source Link -->
-    <IncludeSymbols>true</IncludeSymbols>
-    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
-  </PropertyGroup>
-
```

Comment:
Aha, it seems .NET8 includes these by default now. Nice.

> Starting with .NET 8, Source Link for the following source control providers is included in the .NET SDK and enabled by default

https://github.com/dotnet/sourcelink


However, we target `netstandard2.0` in this project, so I guess we need to keep those? 

Including the packagereference `Microsoft.SourceLink.GitHub` below.

Reply:
The way I understand this is that these settings and packagereference are only used at build/package time, not at runtime, so as long as UnitsNet is build with the .NET 8 SDK it should work with `netstandard2.0`.

I want to validate this but have some trouble. Even if I build the latest master unmodified with the `dotnet build` and `dotnet pack` commands from `Build/build-functions.psm1` the resulting `.nupkg` has missing symbols and compiler flags when checked with https://nuget.info

Reply:
@angularsen any thoughts on this? Am I doing something wrong? Does this only work in CI?

Reply:
@Muximize CI should not do anything different I can think of.

As far as I can tell, the symbols package `.snupkg` includes the `.pdb` file as expected and `.nupkg` includes source code and has valid SourceLink for the commit I had currently checked out: `91a85e2dac60b420fabc44ebde71fc85dab33d6c`

```
git checkout origin/master # Where origin is the original repository
cd UnitsNet
dotnet pack
```

>  UnitsNet -> X:\unitsnet\Artifacts\UnitsNet\net7.0\UnitsNet.dll
>  Successfully created package 'X:\unitsnet\Artifacts/UnitsNet\UnitsNet.5.50.0.nupkg'.
>  Successfully created package 'X:\unitsnet\Artifacts/UnitsNet\UnitsNet.5.50.0.snupkg'.


**Testing the `.nupkg` file on nuget.info**
![image](https://github.com/angularsen/UnitsNet/assets/787816/e15fe839-5fb8-4f28-aea3-7ad17402abce)

**Testing the `.snupkg` symbols file on nuget.info** 
![image](https://github.com/angularsen/UnitsNet/assets/787816/d102f49b-f42e-4347-bb61-233676f148f7)

Reply:
Alright, I finally looked into this. Turns out `IncludeSymbols` and `SymbolPackageFormat` are indeed needed after al, so I restored that but as arguments to `dotnet pack` in `build-functions.psm1`. This also enables sourcelink for the NumberExtensions project without needing per-project settings.

The packagereference is not needed, the symbols and sourcelink are also generated for `netstandard2.0` if build with the .NET 8 SDK.

Reply:
Is it not better to keep it in csproj? Or for consistency, move it to `Directory.Build.props` at the repo root?

This way, it packs nugets the same way regardless of doing it with scripts or built-in tools in IDEs.